### ISSUE TITLE
Do we need to support async requests?

### ISSUE THREAD

**Issue by dshemetov** [cmu-delphi/epidatpy#30]
We have [async request](https://github.com/cmu-delphi/epidatpy/blob/dev/epidatpy/async_request.py) functionality here. Not having worked with async much, I have questions:

* what is the main use case?
* can we do without it?

My guess at the first is: it helps when making many requests in parallel. My guess at the second is: yes, we don't have that functionality in `epidatr` and that client works well enough.

Removing async support would reduce maintenance work and would allow us to switch completely to `delphi-epidata.py` as the backend (see #29).

There is an async smoke test [here](https://github.com/cmu-delphi/epidatpy/blob/dev/smoke_test_async.py) and a regular one [here](https://github.com/cmu-delphi/epidatpy/blob/dev/smoke_test.py). Those could provide some good examples to get started in testing.

**Comment by dshemetov** [comment]
Found [some discussion from a while ago](https://github.com/cmu-delphi/delphi-epidata/pull/1288#issuecomment-1743474822).

**Comment by dshemetov** [comment]
Also [this issue](https://github.com/cmu-delphi/delphi-epidata/issues/1298).

**Comment by dshemetov** [comment]
TL;DR of above issues: 
- async calls were originally intended to get around row limits and pagination in the old PHP server
- current row limits are much larger and we don't do pagination, thus this feature is no longer necessary
- async it will be deprecated in `covidcast` client and eventually downstream code will stop using it
- we can therefore remove this functionality here and simplify the code, this work will happen here #28

**Comment by dshemetov** [comment]
Closed by #28
